### ISSUE TITLE
Missing accessible name for 'Save opportunity' (star icon) button in logged-out state

### ISSUE THREAD

**Issue by eric-serrato95** [HHS/simpler-grants-gov#9563]
### Summary
The same UI element is not accessible when not logged in due to a missing accessible name, which creates an accessibility inconsistency for screen reader users.

### Reproducibility

None

### How to reproduce the bug

Steps to reproduce:

1. Launch URL
2. Go to Search (https://staging.simpler.grants.gov/search)
3. Make sure you are logged out
4. Locate the 'Save opportunity' (star icon) button
5. Inspect the element in DevTools
6. Check accessibility name in the DOM

Actual result:
- The “Save opportunity” button does not have an accessible name (aria-label) when logged out
- The element is rendered as an icon-only button
- <button type="button" class="usa-button usa-button--unstyled" data-testid="button" id="save-search-result-e32a4272-9f3e-4789-ae79-7b62371012c7" data-open-modal="true" aria-controls="simpler-login-modal" fdprocessedid="jr70vo"><svg class="usa-icon button-icon-large" aria-hidden="true" role="img"><use href="/_next/static/media/uswds-sprite.bfef6c15.svg#star_outline"></use></svg></button>

Expected result:
- Button should have an accessible name in all states (logged in and logged out), e.g. 'Save opportunity' or 'Sign in to save opportunity'

### Browser
Chrome, Edge, Firefox

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

<img width="1918" height="990" alt="Image" src="https://github.com/user-attachments/assets/c8d86ed2-8235-4cd3-9ced-79b277d936b8" />

<img width="1916" height="992" alt="Image" src="https://github.com/user-attachments/assets/97b800c0-8db0-444b-8cab-94e080def16c" />

### Acceptance Criteria

- [ ] Confirm with Design the right text for the logged out use case of the star for saved opportunities
- [ ] Respond to the open source PR with updates if needed or test and see if acceptable
- [ ] If open source is not responsive, only then open your own PR

**Comment by eric-serrato95** [comment]
@nkhazi @kkrug UAT ✅ closing ticket!
